CMakeLists.txt

```diff
@@ -36,8 +36,10 @@ endif()
 CHECK_CXX_COMPILER_FLAG(-fuse-ld=gold CXX_SUPPORT_GOLD)
 if(CXX_SUPPORT_GOLD)
     message("-- Using GNU gold linker.")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fuse-ld=gold")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fuse-ld=gold")
+    if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fuse-ld=gold")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fuse-ld=gold")
+    endif()
```

Comment:
```suggestion
    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_C_FLAGS} -fuse-ld=gold")
    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_C_FLAGS} -fuse-ld=gold")
```

Reply:
clang do not support compile options, but support link options.
ok, i will re-commit